python/google/protobuf/pyext/descriptor.cc

```diff
@@ -99,43 +136,64 @@ bool _CalledFromGeneratedFile(int stacklevel) {
   if (frame == nullptr) {
     return false;
   }
+  Py_INCREF(frame);
   while (stacklevel-- > 0) {
-    frame = frame->f_back;
+    PyFrameObject* next_frame = PyFrame_GetBack(frame);
+    Py_DECREF(frame);
+    frame = next_frame;
     if (frame == nullptr) {
       return false;
     }
+    Py_INCREF(frame);
```

Comment:
`PyEval_GetFrame` returns a borrowed reference but `PyFrame_GetBack` returns a strong reference. As a result I think you don't want this last `Py_INCREF(frame)` because `PyFrame_GetBack` will return a string reference with it's refcount already incremented.

Reply:
Valid remark, thanks